Formula/ddcctl.rb

```diff
@@ -0,0 +1,32 @@
+class Ddcctl < Formula
+  desc "DDC monitor controls (brightness) for Mac OSX command-line"
+  homepage "https://github.com/kfix/ddcctl"
+  url "https://github.com/kfix/ddcctl/archive/02dcf6c9ae77783f50a9948a829dee6865131ef6.tar.gz"
+  version "02dcf6c9ae77783f50a9948a829dee6865131ef6"
+  sha256 "e591e14c06b3b8cab2e0b092f8337b60dd6d472c1e2e981352721851fab13085"
+  head "https://github.com/kfix/ddcctl"
+
+  def install
+    if gpu_vendors.find("amd")
+      system "make", "amd"
+    elsif gpu_vendors.find("nvidia")
+      system "make", "nvidia"
+    else
+      system "make", "intel"
+    end
+
+    bin.mkdir
+    cp "ddcctl", "#{bin}/ddcctl"
+  end
+
+  test do
+    system "ddcctl"
```

Comment:
In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.